cni/pkg/install/monitoring.go

```diff
@@ -26,16 +26,10 @@ var (
 	cniInstalls = monitoring.NewSum(
 		"istio_cni_installs_total",
 		"Total number of CNI plugins installed by the Istio CNI installer",
-		monitoring.WithLabels(resultLabel),
```

Comment:
I would not remove this part ( maybe refactor it to avoid the MustCreateLabel - but having a list of labels is very informative, and will still be required if we want monitoring to have a prometheus implementation. 

If we want OTel only ( which I don't mind ) - we don't need an abstraction/wrapper on top of the SDK. But even then it would be good to keep track of what labels are generated, it is probably the most critical aspect of a metric.